rules/DowngradePhp56/Rector/CallLike/DowngradeArgumentUnpackingRector.php

```diff
@@ -0,0 +1,201 @@
+    }
+
+    private function funcCallToCallbackArg(FuncCall $funcCall): Arg
+    {
+        $callback = $funcCall->name instanceof Name ? new String_($funcCall->name->toString()) : $funcCall->name;
+
+        return new Arg($callback);
+    }
+
+    private function methodCallToCallbackArg(MethodCall $methodCall): Arg
+    {
+        $object = $methodCall->var;
+        $method = $methodCall->name instanceof Identifier ? new String_($methodCall->name->toString()) : $methodCall->name;
+        $callback = new Array_([new ArrayItem($object), new ArrayItem($method)]);
+
+        return new Arg($callback);
+    }
+
+    private function staticCallToCallbackArg(StaticCall $staticCall): Arg
+    {
+        if ($staticCall->class instanceof Name) {
+            if ($staticCall->class->isSpecialClassName()) {
+                $class = new String_($staticCall->class->toString());
+            } else {
+                $class = new ClassConstFetch($staticCall->class, 'class');
+            }
+        } else {
+            $class = $staticCall->class;
+        }
+
+        $method = $staticCall->name instanceof Identifier ? new String_($staticCall->name->toString()) : $staticCall->name;
+        $callback = new Array_([new ArrayItem($class), new ArrayItem($method)]);
+
+        return new Arg($callback);
+    }
+
+    /**
+     * @param Arg[] $args
+     */
+    private function createCallUserFuncArrayFuncCall(Arg $callback, array $args): FuncCall
+    {
+        return new FuncCall(new Name('call_user_func_array'), [$callback, $this->mergeArgs($args)]);
+    }
+
+    /**
+     * @param Arg[] $args
+     */
+    private function mergeArgs(array $args): Arg
+    {
+        $unpackedArgs = $this->argManipulator->unpack($args);
+
+        if (count($unpackedArgs) === 1) {
+            return $unpackedArgs[0];
+        }
+
+        return new Arg(new FuncCall(new Name('array_merge'), $unpackedArgs));
+    }
+
+    /**
+     * @param Arg[] $args
+     */
+    private function createReflectionInstantiation(New_ $new, array $args): ?Expr
+    {
+        if ($this->argManipulator->canBeInlined($args)) {
+            $unpackedArgs = $this->argManipulator->unpack($args);
+            Assert::minCount($unpackedArgs, 1);
+            /** @var Array_ $array */
+            $array = $unpackedArgs[0]->value;
+            $arrayItems = array_filter($array->items);
+            $new->args = array_map(fn(ArrayItem $item) => new Arg($item->value), $arrayItems);
+
+            return $new;
+        }
+
+        if ($new->class instanceof Name) {
+            $class = match (strtolower($new->class->toString())) {
+                'self' => new FuncCall(new Name('get_class')),
+                'static' => new FuncCall(new Name('get_called_class')),
+                'parent' => new FuncCall(new Name('get_parent_class')),
+                default => new ClassConstFetch($new->class, 'class'),
+            };
+        } else if ($new->class instanceof Expr) {
```

Comment:
use `elseif` instead of `else if`, you should can run:

```bash
composer fix-cs 
```

to fix cs

Reply:
Is there any reason to not have a CS check in the CI? Would you welcome a PR adding such a thing?

Reply:
@samsonasik For the coding standard, it's not neccessary to include them in code reviews. Rector handles it with our pull-requests or I run them occasionaly in local run.

@villfa You can skip it. The CS was removed from CI, to avoid putting manual work on programmers.